### PR TITLE
Added information to the add-function page

### PR DESCRIPTION
**Pull request by jameshensmancitrix**
Added information describing more paramaters on the devguid/howtos/add-function page. 

Changed the syntax for the example of ~params to use () on tuples to better reflect the codebase

Added details of more parameters used create call

### CODE REVIEW

File: devguide/howtos/add-function.md [hunk 1]
```diff
@@ -63,8 +63,19 @@ The parameters passed to call are all optional (except ~name and ~in_product_sin
   permitted to merely be a single value rather than a list of values. If no
   ~result is specified, the default is unit.
 
+- The ~doc parameter describes what it is, the message is doing.
+
+- The ~hide_from_docs parameter prevents te message from being included in the document ation when generated. This is indicated by a bool.
```
Comment (edwintorok) [inline devguide/howtos/add-function.md:68]: you have an extra space in `document ation`

File: devguide/howtos/add-function.md [hunk 2]
```diff
@@ -63,8 +63,19 @@ The parameters passed to call are all optional (except ~name and ~in_product_sin
   permitted to merely be a single value rather than a list of values. If no
   ~result is specified, the default is unit.
 
+- The ~doc parameter describes what it is, the message is doing.
```
Comment (edwintorok) [inline devguide/howtos/add-function.md:66]: Perhaps:
describes what the message is doing
describes the purpose of the message

File: devguide/howtos/add-function.md [hunk 3]
```diff
@@ -63,8 +63,19 @@ The parameters passed to call are all optional (except ~name and ~in_product_sin
...
+
+- The ~hide_from_docs parameter prevents te message from being included in the document ation when generated. This is indicated by a bool.
+
+- The ~pool_internal parameter is used to indicate if the message should be callable by all or only internal hosts. This is indicated by a bool.
```
Comment (edwintorok) [inline devguide/howtos/add-function.md:70]: It'd be shorter to say  'pool_internal bool parameter'

**Review by edwintorok — COMMENTED** [review]
Please add a Signed-off-By line to your commits, see https://github.com/apps/dco for instructions.
You only need to set this up once (in your ~/.gitconfig for example) and then run `git commit --amend -s` to add the signed-off-by line to the commit message.

File: devguide/howtos/add-function.md (hunk 3 above)
Comment (jameshensmancitrix) [reply devguide/howtos/add-function.md]: fixed

File: devguide/howtos/add-function.md [hunk 4]
```diff
@@ -63,11 +63,11 @@ The parameters passed to call are all optional (except ~name and ~in_product_sin
   permitted to merely be a single value rather than a list of values. If no
   ~result is specified, the default is unit.
 
-- The ~doc parameter describes what it is, the message is doing.
+- The ~doc parameter describles what the message is doing.
```
Comment (edwintorok) [inline devguide/howtos/add-function.md:66]: This introduces a typo in `describles`

File: devguide/howtos/add-function.md [hunk 5]
```diff
@@ -63,8 +63,19 @@ The parameters passed to call are all optional (except ~name and ~in_product_sin
...
+
+- The ~hide_from_docs parameter prevents te message from being included in the document ation when generated. This is indicated by a bool.
+
+- The ~pool_internal parameter is used to indicate if the message should be callable by all or only internal hosts. This is indicated by a bool. 
+
+- The ~errs parameter is a list of possible exceptions that the message can raise.
+
 - The ~allowed_roles parameter is used for access control (see below).
 
+- The parameter ~lifecycle takes in an array of (Status, version, doc) to indicate the lifecycle of message type. This takes over from ~in_oss_since which indicated the release that the message type was introduced in.
```
Comment (psafont) [inline devguide/howtos/add-function.md:76]: For new functions this should be left empty, as there is code will write the correct version it's introduced. You can see this by  running make after adding the functions, the changes will be visible in the file `ocaml/idl/datamodel_lifecycle.ml`

**Review by edwintorok — COMMENTED** [review]
This looks good now, if you squash your commits and add the signed-off-by line to the squashed commit and force push (the way you've added it to your last commit) that should also make the 'DCO check' below pass.

File: devguide/howtos/add-function.md (hunk 5 above)
Comment (jameshensmancitrix) [reply devguide/howtos/add-function.md]: So should I add a sentence to the end explaining that this parameter is added programatically?

File: devguide/howtos/add-function.md [hunk 6]
```diff
@@ -71,7 +71,7 @@ The parameters passed to call are all optional (except ~name and ~in_product_sin
 
 - The ~errs parameter is a list of possible exceptions that the message can raise.
 
-- The parameter ~lifecycle takes in an array of (Status, version, doc) to indicate the lifecycle of the message type. This takes over from ~in_oss_since which indicated the release that the message type was introduced.
+- The parameter ~lifecycle takes in an array of (Status, version, doc) to indicate the lifecycle of the message type. This takes over from ~in_oss_since which indicated the release that the message type was introduced. NOTE: Leave this parmeter empty, it will be populated on build.
```
Comment (edwintorok) [inline devguide/howtos/add-function.md:74]: parameter

**Review by edwintorok — APPROVED** [review]

**Review by psafont — APPROVED** [review]

**Pull request merged**